src/main/perl/FileEditor.pm

```diff
@@ -47,7 +47,15 @@ the class constructor.
 =item new
 
 Returns a new object it accepts the same arguments as the constructor
-for C<CAF::FileWriter>
+for C<CAF::FileWriter> with one additional option: C<source>. This
+option, when present, must be a file name whose contents will be used
+as the initial contents for the edited file if the source modification time
+is more recent than the edited file modification time. This allows to rebuild
+the file contents based on a new version of the reference/template file. 
```

Comment:
`/template` removed but I think that using `source` in the text will be more confusing as one could consider that the source is the filename you passed when instantiating the `FileEditor`. and `reference` as the option name may also be very strange/confusiong... I didn't manage to find a perfect solution! I mentioned in another comment that `source` parameter value is referred internally as reference.